rwmutex_test.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
```

Comment:
This seems odd 😁 

Shouldn't we add a header mentioning that this is based off the `sync` standard library tests ?